opsmngr/blockstore_config.go

```diff
@@ -0,0 +1,152 @@
+// Copyright 2029 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package opsmngr
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	atlas "go.mongodb.org/atlas/mongodbatlas"
+)
+
+const backupAdministratorBlockstoreBasePath = "admin/backup/snapshot/mongoConfigs"
+
+// BlockstoreConfigService is an interface for using the Blockstore Configuration
+// endpoints of the MongoDB Ops Manager API.
+//
+// See more: https://docs.opsmanager.mongodb.com/current/reference/api/admin/backup/blockstore-config/
+type BlockstoreConfigService interface {
+	List(context.Context, *atlas.ListOptions) (*BackupStores, *atlas.Response, error)
+	Get(context.Context, string) (*BackupStore, *atlas.Response, error)
+	Create(context.Context, *BackupStore) (*BackupStore, *atlas.Response, error)
+	Update(context.Context, string, *BackupStore) (*BackupStore, *atlas.Response, error)
+	Delete(context.Context, string) (*atlas.Response, error)
+}
+
+// BackupConfigsServiceOp provides an implementation of the BackupConfigsService interface
+type BlockstoreConfigServiceOp service
+
+var _ BlockstoreConfigService = &BlockstoreConfigServiceOp{}
+
+// BackupStore represents a Blockstore, Oplog and Sync in the MongoDB Ops Manager API
+type BackupStore struct {
+	AdminBackupConfig
+	LoadFactor    int64  `json:"loadFactor,omitempty"`
+	MaxCapacityGB int64  `json:"maxCapacityGB,omitempty"`
```

Comment:
same with bool you need to be careful here as zero (0) won't serialize when `omitempty`, this is not a blocker if you are sure users can't later set this to zero (which in this case won't be able to)